src/Perl6/Actions.nqp

```diff
@@ -1775,30 +1774,17 @@ class Perl6::Actions is HLL::Actions does STDActions {
                     last;
                 }
             }
-        }
```

Comment:
There is no longer a need for `$named_past` and `$longname`. We can get the name from the `CompUnit`, and the originally requested package named is stored in `$target_package`.